linera-views/src/hashable_wrapper.rs

```diff
@@ -61,10 +75,18 @@ where
     }
 
     fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
+        if self.delete_storage_first() {
+            let mut key_prefix = self.inner.context().base_key();
+            key_prefix.pop();
+            batch.delete_key_prefix(key_prefix);
+            self.stored_hash = None;
+        }
         self.inner.flush(batch)?;
```

Comment:
The big issue here is that we do not want to leave anything on the storage after a `clear + flush`. It is not an optimization.

The hash would not be overwritten by the flush in the inner because that would be going outside of its allowed key-space (the arrangement of keys has changed with the use of the Wrapper).

Checking the `batch` that has been created look like a decent solution. It is equally problematic in term of abstraction leakage but it could.

Reply:
I'm not sure it's much of a leakage: Deleting the whole base key (of the inner view, in this case) is pretty much the meaning of "clear", isn't it?

Reply:
> The hash would not be overwritten by the flush in the inner

I was thinking the code after this would then delete the hash. Or would it overwrite it with the hash of the empty view?